sdk/keyvault/Azure.Security.KeyVault.Certificates/tests/LightweightPkcs8DecoderTests.cs

```diff
@@ -0,0 +1,157 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Security.Cryptography;
+using System.Text;
+using Azure.Core;
+using NUnit.Framework;
+
+namespace Azure.Security.KeyVault.Certificates.Tests
+{
+    public class LightweightPkcs8DecoderTests
+    {
+        [Test]
+        public void VerifyECDecoderPrime256v1Imported() =>
+            // The following signature was generated in Key Vault using the certificate key corresponding to EcPrime256v1PrivateKeyImported,
+            // thus proving that Key Vault-generated OID 1.2.840.10045.1.1 (prime-field) is equivalent to OID 1.3.132.0.10 (secp256k1).
+            VerifyECDecoder(EcPrime256v1PrivateKeyImported, 256, @"Dqi/IwHjt2ttGyT0vleMvzsbStSOiDrevV4hhPGAXZoQ55b//keMmZ/weezuiSnliF4bTGWYxrqs73Yoj/6ddQ==");
+
+        [Test]
+        public void VerifyECDecoderSecp256k1() =>
+            VerifyECDecoder(EcSecp256k1PrivateKey, 256, @"YV+z2wZfDRqszLwcoPAnT4XmKMbfAqfwQYdYYU5FyicKWLtRmcaAX8Pd8h0OmSAdxoPO2sVyK2FP22pWq/6fJQ==");
+
+        [Test]
+        public void VerifyECDecoderPrime256v1() =>
+            VerifyECDecoder(EcPrime256v1PrivateKey, 256, @"1Byf+UuKVAEj+lsGBLQ4HiUDd4v/wa/XgcGlOEHorg47kjN2V6EDhC5aLwqZnXkbr48Y1umvBRAWcRYT6f8WfA==");
+
+        [Test]
+        public void VerifyECDecoderSecp384r1() =>
+            VerifyECDecoder(EcSecp384r1PrivateKey, 384, @"gHf3reGX71P7us9/fvcK1+6ONqRnkw5b9v+arz7G8iHk8BGflKwMlESIEEg2EezbNvLg6pURkCPC4OBngnM0OVYCjhb7Onh+zUogi2LSSCEIRZqrYcjzg8PHukp/sRyU");
+
+        [Test]
+        public void VerifyECDecoderSecp521r1() =>
+            VerifyECDecoder(EcSecp521r1PrivateKey, 521, @"ADB74QoZQcAbO0gfOhcavQLhV+nYMfLtNXC0i7a7FOYCiyZIJFoZtEzWbVTe4de1yMoe0lwGnW1mkCoPtEb0Q8f+AYAgRYSXlYkaF96XZYGcSp+RyALtOUpSvvpIaffk/3farwiFlGj6Dqryd5wLRksfLNR4DSyZbpEtHOg2RwkQ+vsI");
+
+        [Test]
+        public void VerifyECDecoderBadData()
+        {
+            byte[] data = Convert.FromBase64String(InvalidEcSecp521r1PrivateKey);
+
+            Assert.Throws<InvalidDataException>(() => LightweightPkcs8Decoder.DecodeECDsaPkcs8(data));
+        }
+
+        // Not using TestCaseSource because params too long for friendly test case rendering.
+        private static void VerifyECDecoder(string key, int keySize, string signature)
+        {
+#if NET461
+            Assert.Ignore("ECC is not supported before .NET Framework 4.7");
+#endif
```

Comment:
Is this valid?  Do you have a separate build of your tests for NET47?

Reply:
Not for Certificates, but for Keys I do because `ECParameters` and friends are not public. Or maybe they are defined but internal? I'd have to dig up runtime assemblies (i.e. non-ref) to find out for sure, but for Keys I had to do all this same work.

Reply:
Per offline conversation, `ECParameters` et. al. is not defined in net461.